components/dashboard/src/settings/EnvironmentVariables.tsx

```diff
@@ -4,12 +4,195 @@
+            </div>
+            <div className="mt-4">
+                <h4>Value</h4>
+                <input className="w-full" type="text" value={ev.value} onChange={(v) => { update({value: v.target.value}) }} />
+            </div>
+            <div className="mt-4">
+                <h4>Scope</h4>
+                <input className="w-full" type="text" value={ev.repositoryPattern} placeholder="org/project" 
+                    onChange={(v) => { update({repositoryPattern: v.target.value}) }} />
+            </div>
+            <div className="mt-3">
+                <p>You can pass a variable for a specific org/project or use wildcard characters (*/*) to make it available in more projects.</p>
+            </div>
+        </div>
+        <div className="flex justify-end mt-6">
+            <button className="text-gray-900 border-white bg-white hover:border-gray-200" onClick={p.onClose}>Cancel</button>
+            <button className={"ml-2 disabled:opacity-50"} onClick={save} >{isNew ? 'Add' : 'Update'} Variable</button>
+        </div>
+    </Modal>
+}
 
 export default function EnvVars() {
-    return <div>
-        <SettingsPage title='Variables' subtitle='Configure environment variables for all workspaces.'>
-            <h3>Environment Variables</h3>
-        </SettingsPage>
-    </div>;
+    const [envVars, setEnvVars] = useState([] as UserEnvVarValue[]);
+    const [currentEnvVar, setCurrentEnvVar] = useState({ name: '', value: '', repositoryPattern: '' } as UserEnvVarValue);
+    const [isAddEnvVarModalVisible, setAddEnvVarModalVisible] = useState(false);
+    const update = async () => {
+        await getGitpodService().server.getEnvVars().then(r => setEnvVars(r));
+    }
+
+    useEffect(() => {
+        update()
+    }, []);
+
+
+    const add = () => {
+        setCurrentEnvVar({ name: '', value: '', repositoryPattern: '' });
+        setAddEnvVarModalVisible(true);
+    }
+
+    const edit = (ev: UserEnvVarValue) => {
+        setCurrentEnvVar(ev);
+        setAddEnvVarModalVisible(true);
+    }
+
+    const save = async (variable: UserEnvVarValue) => {
+        await getGitpodService().server.setEnvVar(variable);
+        await update();
+    };
+
+    const deleteV = async (variable: UserEnvVarValue) => {
+        await getGitpodService().server.deleteEnvVar(variable);
+        await update();
+    };
+
+    const validate = (variable: UserEnvVarValue) => {
+        const name = variable.name;
+        const pattern = variable.repositoryPattern;
+        if (name.trim() === '') {
+            return 'Name must not be empty.';
+        }
+        if (!/^[a-zA-Z0-9_]*$/.test(name)) {
+            return 'Name must match /[a-zA-Z_]+[a-zA-Z0-9_]*/.';
+        }
+        if (variable.value.trim() === '') {
+            return 'Value must not be empty.';
+        }
+        if (pattern.trim() === '') {
+            return 'Scope must not be empty.';
```

Comment:
**praise**: Loved the kumquat colors here! Will get all validations consistent in the next iteration when auditing colors, and more. 🍊